src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -148,6 +149,37 @@ def _construct_line_result(self, output, run_info):
             line_result.output[LINE_NUMBER_KEY] = run_info.index
         return line_result
 
+    @property
+    def has_aggregation_node(self):
+        return hasattr(self, "_aggr_func")
+
+    def _exec_aggregation(
+        self,
+        inputs: List[Any],
+        run_id=None,
+    ) -> AggregationResult:
+        if not self._aggr_func:
+            return AggregationResult({}, {}, {})
+        # Similar to dag flow, add a prefix "reduce" for aggregation run_id.
+        run_id = f"{run_id}_reduce" or f"{str(uuid.uuid4())}_reduce"
+
+        output = None
+        traces = []
+        try:
+            Tracer.start_tracing(run_id)
+            if self._is_async:
```

Comment:
self._is_async only indicates that __call__ is an async function?

Reply:
Refined.